app/test/search/text_utils_test.dart

```diff
@@ -105,7 +105,36 @@ Other useful methods will be added soon...
       });
     });
 
-    test('Cased words', () {
+    test('CamelCase extraction', () {
+      expect(extractCamelCase('CamelCaseWord').toList(),
+          ['Camel', 'Case', 'Word']);
+      expect(extractCamelCase('firstLowerCase').toList(),
+          ['first', 'Lower', 'Case']);
+      expect(extractCamelCase('snake_case_word').toList(), ['snake_case_word']);
+    });
+
+    test('isSelfDocumenting positive examples', () {
+      expect(isSelfDocumenting('childAspectRatio'), true);
+      expect(isSelfDocumenting('coverageScore'), true);
+      expect(isSelfDocumenting('dartdocCoverage'), true);
+      expect(isSelfDocumenting('defaultTodayTextStyle'), true);
+      expect(isSelfDocumenting('iconColor'), true);
+      expect(isSelfDocumenting('maintenanceSuggestions'), true);
+      expect(isSelfDocumenting('todayBorderColor'), true);
+      expect(isSelfDocumenting('updateReport'), true);
+      expect(isSelfDocumenting('weekendTextStyle'), true);
+    });
+
+    test('isSelfDocumenting negative examples', () {
+      expect(isSelfDocumenting('abcdef'), false);
+      expect(isSelfDocumenting('AbcDef'), false);
+      expect(isSelfDocumenting('deriveLookupCandidates'), false);
+      expect(isSelfDocumenting('isSelfDocumenting'), false);
```

Comment:
I would expect this to be considered self-documenting

Reply:
`documenting` is not a frequently used word

Reply:
No - that is my critique of the heuristic. The name 'is' self-documenting (in my opinion), but the predicate says it is not. I don't think matching with frequently used words catches the concept of self-documenting very well.

Reply:
The worst case of these (false-negative) miss-classification is that we still consider a symbol non-documented, while in the real world we understand it perfectly. It doesn't effect the score either way, and if we improve on it in the future, it is a non-breaking experience for the site's users.

I would be more worried if you'd disagree with the positive examples.